python/hail/expr/aggregators.py

```diff
@@ -1163,3 +1161,97 @@ def linreg(y, x):
     k = hl.int32(k)
 
     return _agg_func('LinearRegression', y, t, [k], f=lambda expr: x)
+
+
+@typecheck(group=oneof(anytype, sequenceof(anytype)),
+           agg_expr=expr_any)
```

Comment:
`agg_expr=agg_expr(expr_any)`?

Reply:
It can't be an `agg_expr` because this coerces it to an Aggregable and I want it to be an expression with an IR type of `ApplyAggOp`.